src/nvim/charset.c

```diff
@@ -1743,16 +1743,16 @@ int vim_isblankline(char_u *lbuf)
 /// @param nptr Returns the signed result.
 /// @param unptr Returns the unsigned result.
 void vim_str2nr(char_u *start, int *hexp, int *len, int dooct, int dohex,
-                long *nptr, unsigned long *unptr)
+                NumOpt *nptr, size_t *unptr)
```

Comment:
`vim_str2nr` is used not only for options. I would highly suggest using `intmax_t` and `uintmax_t` instead.